src/ui/public/chrome/directives/app_switcher/app_switcher.js

```diff
@@ -1,19 +1,50 @@
-var parse = require('url').parse;
-var bindKey = require('lodash').bindKey;
+import { parse } from 'url';
+import { bindKey } from 'lodash';
 
 require('../app_switcher/app_switcher.less');
-var DomLocationProvider = require('ui/domLocation');
+const DomLocationProvider = require('ui/domLocation');
 
 require('ui/modules')
 .get('kibana')
+.provider('appSwitcherEnsureNavigation', function () {
+  let forceNavigation = false;
+
+  this.forceNavigation = function (val) {
+    forceNavigation = !!val;
+  };
+
+  this.$get = ['Private', function (Private) {
+    const domLocation = Private(DomLocationProvider);
+
+    return function (event) {
+      if (!forceNavigation || event.isDefaultPrevented() || event.altKey || event.metaKey || event.ctrlKey) {
```

Comment:
Hmm, maybe we could combine these into one nice variable name, or something perhaps to make this look cleaner?